### PR TITLE
feat: [Validation] add `min_dims` rule in FileRules

### PR DESCRIPTION
**Pull request by christianberkman**
**Description**
Add  `min_dims` validation rule to the `FileRules` class

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [X] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

File: system/Validation/FileRules.php [hunk 1]
```diff
@@ -260,4 +260,51 @@ public function max_dims(?string $blank, string $params): bool
...
+     * a specified dimension.
+     */
+    public function min_dims(?string $blank, string $params): bool
+    {
+        // Grab the file name off the top of the $params
+        // after we split it.
+        $params = explode(',', $params);
+        $name   = array_shift($params);
+
+        if (! ($files = $this->request->getFileMultiple($name))) {
```
Comment (kenjis) [inline system/Validation/FileRules.php:275]: Now we accept only **booleans** for if condition.
Please rewrite the condition.

**Review by kenjis — CHANGES_REQUESTED** [review]
The default rule classes locate in `system/Validation/StrictRules/`.
So you need to add this new method in `Validation/StrictRules/FileRules.php`.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Validation/StrictRules/FileRules.php

File: user_guide_src/source/libraries/validation.rst [hunk 2]
```diff
@@ -1082,6 +1082,12 @@ max_dims                Yes         Fails if the maximum width and height of an
                                     the width, and the third is the height. Will
                                     also fail if the file cannot be determined
                                     to be an image.
+min_dims                Yes         Fails if the maximum width and height of an  ``min_dims[field_name,300,150]``
+                                    uploaded image not meet values. The first
+                                    parameter is the field name. The second is
+                                    the width, and the third is the height. Will
+                                    also fail if the file cannot be determined
+                                    to be an image.
```
Comment (kenjis) [inline user_guide_src/source/libraries/validation.rst:1090]: Please add `(This rule was added in v4.6.0.)` at the end.

File: user_guide_src/source/libraries/validation.rst [hunk 3]
```diff
@@ -1082,6 +1082,12 @@ max_dims                Yes         Fails if the maximum width and height of an
                                     the width, and the third is the height. Will
                                     also fail if the file cannot be determined
                                     to be an image.
+min_dims                Yes         Fails if the maximum width and height of an  ``min_dims[field_name,300,150]``
```
Comment (kenjis) [inline user_guide_src/source/libraries/validation.rst:1085]:
```suggestion
min_dims                Yes         Fails if the minimum width and height of an  ``min_dims[field_name,300,150]``
```

**Comment by christianberkman** [comment]
> The default rule classes locate in `system/Validation/StrictRules/`. So you need to add this new method in `Validation/StrictRules/FileRules.php`. https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Validation/StrictRules/FileRules.php

If I understand correctly, I should not add the whole function in `StructRules/FileRules.php` but something alike to this?

```

**Comment by christianberkman** [comment]
> The default rule classes locate in `system/Validation/StrictRules/`. So you need to add this new method in `Validation/StrictRules/FileRules.php`. https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Validation/StrictRules/FileRules.php

Could you confirm to me if i should move or copy the entire method, or should it be a method that checkts the params and calls on the method in the non-strict class? As per now `StrictRules/FileRules.php` is empty so it is hard for me to see what is going on.

**Comment by kenjis** [comment]
Oh, sorry. I didn't look at `StrictRules/FileRules.php` carefully.
It has already the `min_dims()` method.

But there is no test code for the `min_dims()` method in `StrictRules/FileRules.php`.
Could you add test code for it?
and if you could, please refactor two `FileRulesTest` code.

**Comment by christianberkman** [comment]
> Oh, sorry. I didn't look at `StrictRules/FileRules.php` carefully. It has already the `min_dims()` method.
> 

Are we sure we are looking at the same file? I am not finding any methods in `StrictRules/FileRules.php` in my branch, the 4.6 branch or the develop branch? I can only find `min_dims()` in the non strict rules that I added?

**Comment by kenjis** [comment]
I meant that there is the inherited `min_dims()` method in `StrictRules\FileRules`.

File: user_guide_src/source/libraries/validation.rst (hunk 2 above)
Comment (christianberkman) [reply user_guide_src/source/libraries/validation.rst]: Changed

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@christianberkman Thank you!

We don't use merge commits in PR branch. So if you can, use `git rebase` to sync your branch in the next time.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#updating-your-branch

**Comment by christianberkman** [comment]
Thanks for the guidance! Will keep working on my git skills!

**Pull request merged**